File: Algorithms/049_GroupAnagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result;
        map<string, vector<string> > m;
        string temp;
        for (int i=0; i < strs.size(); ++i) {
            temp = strs[i];
            sort(temp.begin(), temp.end());
            if (m.find(temp) == m.end()) {
                vector<string> a;
                a.push_back(strs[i]);
                m[temp] = a;
            } else {
                m[temp].push_back(strs[i]);
            }
        }
        for (map<string, vector<string> >::iterator it=m.begin(); it != m.end(); it++) {
            result.push_back(it->second);
        }
        return result;
    }
};

File: Algorithms/049_GroupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> result = new ArrayList<List<String>>();
        Map<String, ArrayList<String>> m = new HashMap<String, ArrayList<String>>();
        for (int i=0; i < strs.length; ++i) {
            char[] chars = strs[i].toCharArray();
            Arrays.sort(chars);
            String temp = new String(chars);
            if (m.get(temp) == null) {
                ArrayList<String> a = new ArrayList<String>();
                a.add(strs[i]);
                m.put(temp, a);
            } else {
                m.get(temp).add(strs[i]);
            }
        }
        for (List<String> it: m.values()) {
            result.add(it);
        }
        return result;
    }
}
